File: Chapter2/linkedlist_insertion_at_Tail.cpp
#include<iostream>
using namespace std;
class cric{
	public:
		int data;
		cric* next;
		cric(int value){
			data=value;
			next=NULL;
		}
};
void insertCric(cric* &head,int value){
	cric* n=new cric(value);
	if(head==NULL){
		head=n;
		return;
	}
	cric* temp=head;
	while(temp->next!=NULL){
		temp=temp->next;
	}
	temp->next=n;
}
void display(cric* head){
	cric* temp=head;
	while(temp!=NULL){
		cout<<temp->data<<endl;
		temp=temp->next;
	}
}
int main(){
	cric* head=NULL;
	insertCric(head,1);
	insertCric(head,2);
	display(head);
	int n;
	cout<<"enter a value you want insert : ";
	cin>>n;
	insertCric(head,n);
	display(head);
	return 0;
}

File: Chapter2/linkedlist_insertion_at_Tail.py
class Node:
    def __init__(self, d):
        self.d = d
        self.nextval = None


class SLinkedList:
    def __init__(self):
        self.headval = None

    # def insert(self, new_d):
    #     new_node = Node(new_d)
    #     new_node.nextval = self.headval
    #     self.headval = new_node

    def insert(self, new_d):

        # 1. Create a new node
        # 2. Put in the d
        # 3. Set next as None
        new_node = Node(new_d)

        # 4. If the Linked List is empty, then make the
        #    new node as head
        if self.headval is None:
            self.headval = new_node
            return

        # 5. Else traverse till the last node
        last = self.headval
        while (last.nextval):
            last = last.nextval

        # 6. Change the next of last node
        last.nextval = new_node


    def listprint(self):
        printval = self.headval
        while(printval):
            print(f"{printval.d}  ")
            printval = printval.nextval


list = SLinkedList()


list.insert(7)
list.insert(3)
list.insert(6)
list.insert(8)
list.insert(17)

data = int(input("Enter the d to insert"))
list.insert(d)

list.listprint()
